src/gpu/metal/SDL_gpu_metal.m

```diff
@@ -0,0 +1,3977 @@
+    for (i = 0; i < commandBuffer->count; i += 1) {            \
+        if (commandBuffer->array[i] == resource) {             \
+            return;                                            \
+        }                                                      \
+    }                                                          \
+                                                               \
+    if (commandBuffer->count == commandBuffer->capacity) {     \
+        commandBuffer->capacity += 1;                          \
+        commandBuffer->array = SDL_realloc(                    \
+            commandBuffer->array,                              \
+            commandBuffer->capacity * sizeof(type));           \
+    }                                                          \
+    commandBuffer->array[commandBuffer->count] = resource;     \
+    commandBuffer->count += 1;                                 \
+    SDL_AtomicIncRef(&resource->referenceCount);
+
+/* Blit Shaders */
+
+static const char *FullscreenVertexShader =
+    "using namespace metal;\n"
+    "struct VertexToPixel { float4 position [[position]]; float2 texcoord; };\n"
+    "vertex VertexToPixel vs_main(uint vI [[vertex_id]]) {\n"
+    "   float2 inTexcoord = float2((vI << 1) & 2, vI & 2);\n"
+    "   VertexToPixel out;\n"
+    "   out.position = float4(inTexcoord * float2(2.0f, -2.0f) + float2(-1.0f, 1.0f), 0.0f, 1.0f);\n"
+    "   out.texcoord = inTexcoord;\n"
+    "   return out;\n"
+    "}";
+
+static const char *BlitFrom2DFragmentShader =
+    "using namespace metal;\n"
+    "struct VertexToPixel { float4 position [[position]]; float2 texcoord; };\n"
+    "fragment float4 fs_main(\n"
+    "   VertexToPixel input [[stage_in]],\n"
+    "   texture2d<float> srcTexture [[texture(0)]],\n"
+    "   sampler srcSampler [[sampler(0)]]) {\n"
+    "   return srcTexture.sample(srcSampler, input.texcoord);\n"
+    "}";
+
+/* Forward Declarations */
+
+static void METAL_Wait(SDL_GpuRenderer *driverData);
+static void METAL_UnclaimWindow(
+    SDL_GpuRenderer *driverData,
+    SDL_Window *window);
+static void METAL_INTERNAL_DestroyBlitResources(SDL_GpuRenderer *driverData);
+
+/* Conversions */
+
+static MTLPixelFormat SDLToMetal_SurfaceFormat[] = {
+    MTLPixelFormatRGBA8Unorm,   /* R8G8B8A8 */
+    MTLPixelFormatBGRA8Unorm,   /* B8G8R8A8 */
+    MTLPixelFormatB5G6R5Unorm,  /* B5G6R5 */
+    MTLPixelFormatBGR5A1Unorm,  /* B5G5R5A1 */
+    MTLPixelFormatABGR4Unorm,   /* B4G4R4A4 */
+    MTLPixelFormatRGB10A2Unorm, /* A2R10G10B10 */
+    MTLPixelFormatRG16Unorm,    /* R16G16 */
+    MTLPixelFormatRGBA16Unorm,  /* R16G16B16A16 */
+    MTLPixelFormatR8Unorm,      /* R8 */
+    MTLPixelFormatA8Unorm,      /* A8 */
+#ifdef SDL_PLATFORM_MACOS
+    MTLPixelFormatBC1_RGBA,      /* BC1 */
+    MTLPixelFormatBC2_RGBA,      /* BC2 */
+    MTLPixelFormatBC3_RGBA,      /* BC3 */
+    MTLPixelFormatBC7_RGBAUnorm, /* BC7 */
+#else
+    MTLPixelFormatInvalid, /* BC1 */
```

Comment:
These are supported on newer iOS versions, depending on the GPU.

Reply:
I think this came down to SDK versioning errors, but I don't know the preprocessor magic to make this work for our minimum (macOS is 11.x, right? Not sure how that lines up with iOS)